src/lib/support/Variant.h

```diff
@@ -225,6 +225,10 @@ struct Variant
         return *reinterpret_cast<const T *>(&mData);
     }
 
+    // TODO this is not ideal, while we prevent the Optional version of this from causing an issue we could be suppressing real
+    // errors.
+
+    // NOLINTNEXTLINE(clang-analyzer-core.CallAndMessage): When ~Value() is called the mData.~T(); is not called
```

Comment:
TODO comment is a little off. That is because I am still hoping that we can figure something out in Optional.h to suppress this clang warning. Still seeing how we can do that